uaclient/tests/test_auto_attach.py

```diff
@@ -0,0 +1,132 @@
+import contextlib
```

Comment:
nitpick, can we name the file `test_lib_auto_attach.py` to make it more clear that this is testing the file in the lib directory.